presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -250,6 +250,7 @@ public void flushCache()
         tableStatisticsCache.invalidateAll();
         partitionStatisticsCache.invalidateAll();
         rolesCache.invalidateAll();
+        roleGrantsCache.invalidateAll();
```

Comment:
This does not look like related to `GRANT/REVOKE`. Please extract this as separate commit.

Reply:
will split into 3 commits a moment before merge, doing this on every fix is clumsy

Reply:
And what about: grantedPrincipalsCache (my bad), and grantedPrincipalsCache?

Reply:
in fact 4 caches weren't cleaned here :-(
I added cleaning in a loop to solve this once and for all